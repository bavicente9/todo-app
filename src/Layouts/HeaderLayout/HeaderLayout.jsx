import './HeaderLayout.scss'
import { useDispatch, useSelector } from 'react-redux';
import {selectTheme, toggleTheme} from '../../features/theme/themeSlice'


const HeaderLayout = () => {

    const theme = useSelector(selectTheme)
    const dispatch = useDispatch()

    const handleToggleTheme = () => {
      dispatch(toggleTheme())
    }
    return(
        <header className='header'>
            <div className='header_container'>
                <h1 className='title'>TODO</h1>
                <button
                  className={`buttonTheme buttonTheme--${theme}`} 
                    onClick={handleToggleTheme}
                    type='button'
                  />
            </div>
        </header>
    )
}

export default HeaderLayout;