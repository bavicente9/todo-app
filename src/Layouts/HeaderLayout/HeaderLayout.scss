@import '../../Utils/style/globalVariables.scss';


.header{
    padding-top: 40px;
    .header_container{

        
        min-width: $globalContainer-minWidth;
        max-width: $globalContainer-maxWidth;
        margin: 0 auto;
        display: flex;
        flex-wrap: nowrap;
        
        justify-content: space-between;
        align-items: center;
        
        color: #fff;
        letter-spacing: 5px;

        .buttonTheme{
            border: none;
            background: none;

            height: 25px;
            width: 25px;
            background-repeat: none;
            background-size: cover;
            cursor: pointer;
        }
        
        .buttonTheme--dark{
            background-image: url('../../Assets/darkMode/icon-sun.svg');
        }
        .buttonTheme--light{
            background-image: url('../../Assets/lightMode/icon-moon.svg');
        }
        
        
    }
    
    @media (max-width:$globalMedia-maxWidth) {
        padding-top: 20px;
        .header_container{
           min-width: $globalMobileContainer-minWidth;
           max-width: $globalMobileContainer-maxWidth;

       }
    }
}