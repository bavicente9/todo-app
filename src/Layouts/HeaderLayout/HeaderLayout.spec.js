import renderer from 'react-test-renderer';
import HeaderLayout from "./HeaderLayout"
import componentWithMockStore from "../../Utils/test/componentWithMockStore";




describe('HeaderLayout', () => {
    let store
    let component;
    beforeEach(() => {
        //render a component with a mock Store
        ( {component, store} = componentWithMockStore(<HeaderLayout />  ))
    });

    it('match with snapshot', () => {
        expect(component.toJSON()).toMatchSnapshot();
    })


    it('button theme dispatch a action on button click', () => {

        renderer.act(() => {
            component.root.findByType('button').props.onClick();
        })

        expect(store.dispatch).toHaveBeenCalledTimes(1);
    })
 
})