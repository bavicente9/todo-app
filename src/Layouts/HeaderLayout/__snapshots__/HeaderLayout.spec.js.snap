// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLayout match with snapshot 1`] = `
<header
  className="header"
>
  <div
    className="header_container"
  >
    <h1
      className="title"
    >
      TODO
    </h1>
    <button
      className="buttonTheme buttonTheme--dark"
      onClick={[Function]}
      type="button"
    />
  </div>
</header>
`;
